### ISSUE TITLE
bean-format fails on currencies that contain "_" (underscore)

### ISSUE THREAD

**Issue by blais** [beancount/beancount#146]
**[Original report](https://bitbucket.org/blais/beancount/issue/146) by Stefano Zacchiroli (Bitbucket: [zacchiro](https://bitbucket.org/zacchiro), GitHub: [zacchiro](https://github.com/zacchiro)).**

----------------------------------------

The following beancount input:
```
#!beancount

1970-01-01 open Equity:Opening-balances
1970-01-01 open Assets:Investments

2014-03-31 * "opening"
  Assets:Investments                 1.23 FOO_BAR
  Equity:Opening-balances
```
passes bean-check just fine, but makes bean-format fail with an assertion error:
```
#!

Traceback (most recent call last):
  File "/home/zack/.local/bin/bean-format", line 2, in <module>
    from beancount.scripts.format import main; main()
  File "/home/zack/.local/lib/python3.5/site-packages/beancount/scripts/format.py", line 154, in main
    formatted_contents = align_beancount(contents)
  File "/home/zack/.local/lib/python3.5/site-packages/beancount/scripts/format.py", line 81, in align_beancount
    assert (old_stripped == new_stripped), (old_stripped, new_stripped)
AssertionError: ('1970-01-01 open Equity:Opening-balances 1970-01-01 open Assets:Investments 2014-03-31 * "opening" Assets:Investments 1.23 FOO_BAR Equity:Opening-balances', '1970-01-01 open Equity:Opening-balances 1970-01-01 open Assets:Investments 2014-03-31 * "opening" Assets:Investments Equity:Opening-balances')
```
By the look of it, it's possible that just an "_" is missing in the final part of the regex in python/beancount/scripts/format.py:align_beancount(). But I haven't tested the fix yet.

**Comment by blais** [comment]
**Original comment by Stefano Zacchiroli (Bitbucket: [zacchiro](https://bitbucket.org/zacchiro), GitHub: [zacchiro](https://github.com/zacchiro)).**

----------------------------------------

I forgot to mention: the above is the minimal example on which I can reproduce this issue, and removing the "_" from the currency name makes the issue go away (hence the hypothesis in the bug title :))

**Comment by blais** [comment]
**Original comment by Stefano Zacchiroli (Bitbucket: [zacchiro](https://bitbucket.org/zacchiro), GitHub: [zacchiro](https://github.com/zacchiro)).**

----------------------------------------

I confirm that adapting the regex I mentioned fixes the problem.
Here is a patch that would close this bug.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #146 and more: Made bean-format depend on the centrally defined regular expressions.
